### PR TITLE
[MU4] Multiprojects. Added sync changes settings between instances 

### PR DESCRIPTION
**Pull request by igorkorsukov**
https://user-images.githubusercontent.com/3818029/122218369-abfc4480-ceae-11eb-9582-0dad303e9589.mp4

### CODE REVIEW

File: src/multiinstances/internal/multiinstancesprovider.cpp [hunk 1]
```diff
@@ -73,6 +83,28 @@ void MultiInstancesProvider::onMsg(const Msg& msg)
             mainWindow()->requestShowOnFront();
         }
     }
+    // Settings
+    else if (msg.type == MsgType::Request && msg.method == METHOD_PREFERENCES_IS_OPENED) {
+        bool isOpened = interactive()->isOpened(PREFERENCES_URI).val;
+        m_ipcChannel->response(METHOD_PREFERENCES_IS_OPENED, { QString::number(isOpened) }, msg.srcID);
+    } else if (msg.method == METHOD_SETTINGS_BEGIN_TRANSACTION) {
```
Comment (cbjeukendrup) [inline src/multiinstances/internal/multiinstancesprovider.cpp:90]: Wrong key? Should be METHOD_ACTIVATE_WINDOW_WITH_OPENED_PREFERENCES, I think.

Comment (igorkorsukov) [reply src/multiinstances/internal/multiinstancesprovider.cpp]: big thanks, fixed, hmm, before making PR, I tested..., hmm

**Review by vpereverzev — APPROVED** [review]

File: src/framework/global/settings.cpp [hunk 2]
```diff
@@ -110,6 +116,10 @@ Settings::Items Settings::readItems() const
         result[item.key] = item;
     }
 
+    if (multiInstancesProvider()) {
```
Comment (vpereverzev) [inline src/framework/global/settings.cpp:119]: BTW it might make sense to put IF_ASSERT here and below

Comment (igorkorsukov) [reply src/framework/global/settings.cpp:119]: No, just ifs are needed here, and I think it's better to keep them. They are needed in tests, in tests, there is a call to the settings.

**Pull request merged**